bridges/snowbridge/pallets/system-frontend/src/lib.rs

```diff
@@ -198,16 +205,18 @@ pub mod pallet {
 	}
 
 	impl<T: Config> Pallet<T> {
-		pub fn send(xcm: Xcm<()>) -> DispatchResult {
-			let bridgehub = Location::new(1, [Parachain(1002)]);
-			send_xcm::<T::XcmSender>(bridgehub, xcm).map_err(|_| Error::<T>::Send)?;
+		pub fn send(origin: Location, xcm: Xcm<()>) -> DispatchResult {
+			let bridgehub = T::BridgeHub::get();
+			let (_, price) =
+				send_xcm::<T::XcmSender>(bridgehub, xcm).map_err(|_| Error::<T>::Send)?;
+			T::XcmExecutor::charge_fees(origin, price).map_err(|_| Error::<T>::FundsUnavailable)?;
 			Ok(())
 		}
 		pub fn burn_fees(origin_location: Location, fee: u128) -> DispatchResult {
 			let ethereum_fee_asset = (T::FeeAsset::get(), fee).into();
-			T::AssetTransactor::withdraw_asset(&ethereum_fee_asset, &origin_location, None)
+			T::XcmExecutor::charge_fees(origin_location.clone(), ethereum_fee_asset)
 				.map_err(|_| Error::<T>::FundsUnavailable)?;
-			T::AssetTransactor::withdraw_asset(&T::DeliveryFee::get(), &origin_location, None)
+			T::XcmExecutor::charge_fees(origin_location, T::RemoteExecutionFee::get().into())
 				.map_err(|_| Error::<T>::FundsUnavailable)?;
 			Ok(())
 		}
```

Comment:
This whole function is problematic.

Firstly, you cannot initiate a teleport (burning) by using `T::XcmExecutor::charge_fees`. That does not burn, it simply transfers the value to a treasury account. Rather use `T::AssetTransactor::withdraw_asset` like you were doing.

Secondly, the function is confusing, because its signature is `burn_fees(location, fee)`, which implies that only `fee` is being being burnt. However the function is also secretly burning `T::RemoteExecutionFee::get()`, which is confusing for reviewers.

Rather create a generic `burn_for_teleport(location, asset)` function that is called twice:

```rust
burn_for_teleport(origin_location, ether_fee);
burn_for_teleport(origin_location, remote_execution_fee);
```

Reply:
Nice catch! 
https://github.com/vgeddes/polkadot-sdk/pull/6/commits/c9369a615cb7f75bc50177db1b0fc12048b425e5 for the fix.

Btw: For less confusion I'v removed the  `burn_fees` function. Extracting `burn_for_teleport` also seems unnecessary since it contains only a single line of code, so I prefer to inline it instead.

Reply:
I prefer factoring out the burning into `burn_for_teleport`.

Also you must also call `AssetTransactor::can_check_out`, like what we had to for V1.

Reply:
https://github.com/vgeddes/polkadot-sdk/pull/6/commits/af8205340d5860f51e3dc49606471320084fc96b